daliuge-engine/dlg/apps/dockerapp.py

```diff
@@ -315,6 +320,20 @@ def run(self):
         binds = list(set(binds))   # make this a unique list else docker complains
         logger.debug("Volume bindings: %r", binds)
 
+        portMappings = {}  # = {'5005/tcp': 5005, '5006/tcp': 5006}
+        for mapping in self._portMappings.split(','):
```

Comment:
its worth noting this is exposing containers to public networks and a secure connection should be used. There are supposedly other possibilities for network binding such as specifying a private network ip (e.g. `docker run -p localhost:5005:5005` or `docker run --net=<custom bridge network>`)